### PR TITLE
Fix quotes for configuration options

### PR DESCRIPTION
**Pull request by nicolasparada**
Fixes an issue with quoted strings in config.
They are not unquoted when reading, causing unexpected results. For example, on the http_loader you can use multiline on the yaml format, when converting it back to classic format, everything gets inlined inside double quotes, and quotes inside the template functions get escaped. This causes template execution to fail. This PR should fix it.

Important. Looks like the tests were not working as expected and were not asserting anything. The output was split by `\n` and then for each line was doing an assertion. But didn't check that there were no assertions...
Now they run, but noticed that the output plugins fails with an error without enough details. Could not fix it on this PR, so for the test I stopped testing the output plugin and used a builtin fluent-bit plugin `file` which does exactly what the output plugin in the tests does.
We will have to fix this problem.
The error was just:

```
fatal: morestack on g0
```

The output plugin error is unrelated to this PR so this should be good to merge.

### CODE REVIEW

**Comment by nicolasparada** [comment]
I need help, there is a `testate/Dockerfile` which runs fluent-bit with an output plugin and we mount a temporal file into `/fluent-bit/etc/output.txt` but the output plugin `file` is giving `Permission Denied` error on the file path `/fluent-bit/etc/output.txt`.

I don't know how to fix this.

File: testdata/Dockerfile [hunk 1]
```diff
@@ -19,6 +22,10 @@ RUN go mod verify
 
 COPY . .
 
+RUN rm -rf ./testdata/fluent-bit.conf
+RUN rm -rf ./testdata/plugins.conf
+RUN rm -rf ./testdata/output.txt
```
Comment (niedbalski) [inline testdata/Dockerfile:27]: Why is this required? I don't think it is.

File: plugin_test.go [hunk 2]
```diff
@@ -2,37 +2,40 @@ package plugin
...
-	wantNoErr(t, err)
+	assert.NoError(t, err)
 
 	//nolint:gosec //required filesystem access to read fixture data.
-	f, err := os.Create(filepath.Join(pwd, "testdata/output.txt"))
-	wantNoErr(t, err)
-
-	t.Cleanup(func() { _ = f.Close() })
-	t.Cleanup(func() { _ = f.Truncate(0) })
+	f, err := os.Create(filepath.Join(t.TempDir(), "output.txt"))
```
Comment (niedbalski) [inline plugin_test.go:37]: What are you doing with this?

**Review by niedbalski — CHANGES_REQUESTED** [review]
Minor fix required. Indeed it works on mac. I will review this a bit deeper

```
➜ plugin (main) ✔ go test -v ./...
?   	github.com/calyptia/plugin/input	[no test files]
?   	github.com/calyptia/plugin/metric	[no test files]
?   	github.com/calyptia/plugin/metric/cmetric	[no test files]
=== RUN   TestPlugin
--- PASS: TestPlugin (101.05s)
PASS
ok  	github.com/calyptia/plugin	102.122s
=== RUN   TestGetRecord
--- PASS: TestGetRecord (0.00s)
PASS
ok  	github.com/calyptia/plugin/output	(cached)
```

**Review by niedbalski — APPROVED** [review]
LGTM

File: plugin_test.go [hunk 3]
```diff
@@ -37,6 +37,17 @@ func testPlugin(t *testing.T, pool *dockertest.Pool) {
 	f, err := os.Create(filepath.Join(t.TempDir(), "output.txt"))
 	assert.NoError(t, err)
 
+	defer func() {
+		err := os.RemoveAll(f.Name())
```
Comment (nicolasparada) [inline plugin_test.go:41]: This is not required since I was using `t.TempDir()` and that entire directory gets removed once the test finishes.

**Pull request merged**